frontend/packages/dev-console/src/components/import/serverless/ServerlessSection.tsx

```diff
@@ -0,0 +1,42 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import * as _ from 'lodash';
+import { stateToProps } from '@console/internal/reducers/features';
+import { FLAG_KNATIVE_SERVING } from '@console/knative-plugin/src/const';
+import FormSection from '../section/FormSection';
+import { CheckboxField } from '../../formik-fields';
+import './ServerlessSection.scss';
+
+type ServerlessSectionProps = {
+  flags: { [key: string]: boolean };
+};
+
+const ServerlessSection: React.FC<ServerlessSectionProps> = ({ flags }) => {
+  if (flags[FLAG_KNATIVE_SERVING]) {
+    return (
+      <FormSection title="Serverless Options" divider>
+        <button
+          title="Tech Preview"
+          type="button"
+          className="btn pf-c-button odc-serverless-tech-preview"
+        >
+          Tech Preview
+        </button>
+        <CheckboxField
+          type="checkbox"
+          label="Enable scaling to zero when idle"
+          name="serverless.trigger"
+        />
+      </FormSection>
+    );
+  }
+
+  return null;
+};
+
+export default connect(
+  (state) => stateToProps([FLAG_KNATIVE_SERVING], state),
```

Comment:
Can you use `connectToFlags` to avoid `areStatePropsEqual`?

Reply:
if i have to use it, i have to modify the type definition of `connectToFlag`. I believe you and @invincibleJai already have discussed about this.

Reply:
it is also just a wrapper around the exact same call as is here

Reply:
Lots of places use `connectToFlags`. What's the issue you have with it and why do you need to modify it's types?

Reply:
`connectToFlags` expects the type of Flag enum as param. I was told we can't add knative serving into the enum. So I have to change the type of `connectToFlags` function param to accept string too. @joshuawilson has already done that. I have sync'd with his PR.